internal/build/tar.go

```diff
@@ -232,6 +233,27 @@ func (a *ArchiveBuilder) writeEntry(entry archiveEntry) error {
 		_ = file.Close()
 	}()
 
+	// We want to balance two needs:
+	//
+	// 1) Don't blow up heap size when copying large binary blobs.
+	// 2) If there's a race condition (e.g., a user is actively
+	//    writing to a file while we tar it up), we want to make sure
+	//    there's not a mismatch between the size of the file contents
+	//    in the tar, and the size we declare in the header.
+	//
+	// So for sufficiently small files, we copy the file into an in-memory buffer
+	// and use the size as the source of truth.
```

Comment:
It's not super clear to me why we're doing this or what happens for large files. Is something like this accurate?

```
// The size header must match the number of bytes. There is room for a race condition here if something writes to the file after we've read the file size.
// For small files, we avoid this by first copying the file into a buffer.
// For larger files, we don't want to copy the whole thing into a buffer, so we just let the error happen.
```

Reply:
added this to the comment!